modules/flowable-engine/src/test/java/org/flowable/engine/test/tenant/ChangeTenantIdProcessTest.java

```diff
@@ -0,0 +1,590 @@
+
+        // All the instances should now be assigned to the tenant B
+        checkTenantIdForAllInstances(processInstancesNoTenant, TEST_TENANT_B, "after the change to " + TEST_TENANT_B);
+        checkTenantIdForAllInstances(processInstancesTenantB, TEST_TENANT_B, "after the change to " + TEST_TENANT_B);
+
+        // The instances for tenant C remain untouched
+        checkTenantIdForAllInstances(processInstancesTenantC, TEST_TENANT_C, "after the change to " + TEST_TENANT_B);
+
+        //Expected results map
+        Map<String, Long> resultMap = new HashMap<>();
+        resultMap.put(BpmnChangeTenantIdEntityTypes.ACTIVITY_INSTANCES, 35L);
+        resultMap.put(BpmnChangeTenantIdEntityTypes.EXECUTIONS, 16L);
+        resultMap.put(BpmnChangeTenantIdEntityTypes.EVENT_SUBSCRIPTIONS, 2L);
+        resultMap.put(BpmnChangeTenantIdEntityTypes.TASKS, 1L);
+        resultMap.put(BpmnChangeTenantIdEntityTypes.EXTERNAL_WORKER_JOBS, 1L);
+        resultMap.put(BpmnChangeTenantIdEntityTypes.HISTORIC_ACTIVITY_INSTANCES, 44L);
+        resultMap.put(BpmnChangeTenantIdEntityTypes.HISTORIC_PROCESS_INSTANCES, 4L);
+        resultMap.put(BpmnChangeTenantIdEntityTypes.HISTORIC_TASK_LOG_ENTRIES, 7L);
+        resultMap.put(BpmnChangeTenantIdEntityTypes.HISTORIC_TASK_INSTANCES, 4L);
+        resultMap.put(BpmnChangeTenantIdEntityTypes.JOBS, 1L);
+        resultMap.put(BpmnChangeTenantIdEntityTypes.SUSPENDED_JOBS, 5L);
+        resultMap.put(BpmnChangeTenantIdEntityTypes.TIMER_JOBS, 2L);
+        resultMap.put(BpmnChangeTenantIdEntityTypes.DEADLETTER_JOBS, 1L);
+
+        //Check that all the entities are returned
+        assertThat(simulationResult.getChangedEntityTypes())
+                .containsExactlyInAnyOrderElementsOf(resultMap.keySet());
+        assertThat(result.getChangedEntityTypes())
+                .containsExactlyInAnyOrderElementsOf(resultMap.keySet());
+
+        resultMap.forEach((key, value) -> {
+            //Check simulation result content
+            assertThat(simulationResult.getChangedInstances(key))
+                    .as(key)
+                    .isEqualTo(value);
+
+            //Check result content
+            assertThat(result.getChangedInstances(key))
+                    .as(key)
+                    .isEqualTo(value);
+        });
+
+        //Check that we can complete the active instances that we have changed
+        completeTask(processInstanceIdNoTenantActive);
+        assertProcessEnded(processInstanceIdNoTenantActive);
+        completeTask(processInstanceIdBActive);
+        assertProcessEnded(processInstanceIdBActive);
+        completeTask(processInstanceIdCActive);
+        assertProcessEnded(processInstanceIdCActive);
+
+        assertThat(eventListener.events).hasSize(1);
+        FlowableChangeTenantIdEvent event = eventListener.events.get(0);
+        assertThat(event.getEngineScopeType()).isEqualTo(ScopeTypes.BPMN);
+        assertThat(event.getSourceTenantId()).isEqualTo("");
```

Comment:
isn't `isEmpty()` clearer?

Reply:
Not sure. Now I can search for `""` and find its usages in this test.